modules/mod-deep-learning/ModelCardPanel.cpp

```diff
@@ -0,0 +1,517 @@
+/**********************************************************************
+
+   Audacity: A Digital Audio Editor
+   Audacity(R) is copyright (c) 1999-2021 Audacity Team.
+
+   ModelCardPanel.cpp
+   Hugo Flores Garcia
+
+******************************************************************/
+
+#include "ModelCardPanel.h"
+#include "DeepLearningEffectBase.h"
+#include "effects/EffectUI.h"
+
+#include "Shuttle.h"
+#include "ShuttleGui.h"
+
+#include <wx/scrolwin.h>
+#include <wx/range.h>
+#include <wx/gauge.h>
+#include <wx/button.h>
+#include <wx/sizer.h>
+#include <wx/textdlg.h>
+#include <wx/stattext.h>
+#include <wx/hyperlink.h>
+
+#include "Internat.h"
+#include "AllThemeResources.h"
+#include "Theme.h"
+
+// DomainTagPanel
+
+class DomainTagPanel : public wxPanelWrapper
+{
+public:
+   DomainTagPanel(wxWindow *parent, const wxString &tag, const wxColour &color)
+                  :wxPanelWrapper(parent, wxID_ANY, wxDefaultPosition, wxDefaultSize)
+   {
+      TranslatableString name = Verbatim("%s").Format(tag);
+      SetLabel(name);
+
+      // SetWindowStyle(wxBORDER_SIMPLE);
+      SetBackgroundColour(color);
+
+      ShuttleGui S(this, eIsCreating);
+      wxStaticText *txt = S.AddVariableText(name, true);
+      SetMaxSize(wxSize(90, static_cast<int>(txt->GetSize().GetHeight() * 2)));
+      SetVirtualSize(txt->GetSize());
+
+      txt->SetBackgroundColour(color);
+      wxFont font = txt->GetFont();
+      font.SetPointSize(11);
+      txt->SetFont(font);
+
+      Refresh();
+      
+      Fit();
+      Layout();
+   }
+};
+
+// ModelCardPanel
+
+ModelCardPanel::ModelCardPanel(wxWindow *parent, wxWindowID winid, ModelCardHolder card, 
+                              DeepLearningEffectBase *effect, ModelManagerPanel *managerPanel, const wxSize& size)
+    : wxPanelWrapper(parent, winid, wxDefaultPosition, size, wxBORDER_SIMPLE | wxEXPAND), 
+      mModelName(nullptr),
+      mModelSize(nullptr),
+      mModelAuthor(nullptr), 
+      mShortDescription(nullptr),  
+      mLongDescription(nullptr),  
+      mInstallButton(nullptr), 
+      mInstallStatusText(nullptr),
+      mInstallProgressGauge(nullptr),
+      mSelectButton(nullptr), 
+      mMoreInfoButton(nullptr)
```

Comment:
You omitted `mApplyButton` !

Again, in-class initializers for the pointers would make it easier to check that there is no omission.

Reply:
525fb9b9e497bcd38a812067cf7ce69350e9bcec